base/tks/src/main/java/org/dogtagpki/server/tks/servlet/TokenServlet.java

```diff
@@ -2815,10 +2883,28 @@ private void processComputeSessionKeysSCP03(HttpServletRequest req, HttpServletR
         }
 
         if (serversideKeygen == true) {
-            if (serverSideValues.size() == 3) {
+            logger.debug(method + " serversideValues.size: " + serverSideValues.size());
+	    for(int i = 0 ; i < serverSideValues.size(); i++) {
+                logger.debug(method + " val:" + i  + " : " + serverSideValues.get(i));
+            }
+
+            if (serverSideValues.size() >= 3) {
+                logger.debug(method + " size >= 3 ");
                 drm_trans_wrapped_desKeyString = serverSideValues.get(2);
                 kek_wrapped_desKeyString = serverSideValues.get(0);
                 keycheck_s = serverSideValues.get(1);
+                keycheck_aes_s = serverSideValues.get(2);
```

Comment:
so drm_trans_wrapped_desKeyString and keycheck_aes_s are the same value?  This is very confusing.

Reply:
Will fix this, it's really get(0) for drm_trans_wrapped_desKeyString.